server/src/main/java/org/opensearch/index/shard/StoreRecovery.java

```diff
@@ -491,9 +497,22 @@ private void recoverFromRemoteStore(IndexShard indexShard) throws IndexShardReco
                 }
             }
 
-            // This creates empty trans-log for now
-            // ToDo: Add code to restore from remote trans-log
-            bootstrap(indexShard, store);
+            if (repository != null) {
+                FileTransferTracker fileTransferTracker = new FileTransferTracker(shardId);
+                assert repository instanceof BlobStoreRepository : "repository should be instance of BlobStoreRepository";
+                BlobStoreRepository blobStoreRepository = (BlobStoreRepository) repository;
+                TranslogTransferManager translogTransferManager = new TranslogTransferManager(
+                    new BlobStoreTransferService(
+                        blobStoreRepository.blobStore(),
+                        indexShard.getThreadPool().executor(ThreadPool.Names.TRANSLOG_TRANSFER)
+                    ),
+                    blobStoreRepository.basePath().add(shardId.getIndex().getUUID()).add(String.valueOf(shardId.id())),
+                    fileTransferTracker,
+                    fileTransferTracker::exclusionFilter
+                );
```

Comment:
should we move this to `RemoteFSTranslog#download` ? Creating `translogTransferManager`, `fileTransferTracker`  etc is responsibility of `RemoteFSTranslog` only . 

We can then try to move the common parts of the ctor and `download` into separate functions.

Reply:
The problem is `RemoteFsTranslog#download` is static method. Even if we create the instances of tracker in download method, we will not be able to re-use it in `RemoteFsTranslog`.

Reply:
I agree on that , but at least the logic of creation of `fileTransferTracker`, `translogTransferManager` will reside in `RemoteFSTranslog` only . We will not be reuse the instances, but reuse the code .

Reply:
In this case, do we assume that `FileTransferTracker` and `TranslogTransferManager` should not be initialized outside RemoteFsTranslog? Currently, they are public classes so can be instantiated outside.